### ISSUE TITLE
Detonations of EBS related exfil techniques fails when EBS encryption by default is enabled for a region

### ISSUE THREAD

**Issue by mchaffe** [DataDog/stratus-red-team#109]
**What is not working?**
Detonation fails for `aws.exfiltration.ec2-share-ami` and `aws.exfiltration.ec2-share-ebs-snapshot` when EBS encryption by default is enabled for a region

**What OS are you using?**
N/A

**What is your Stratus Red Team version?**
`stratus version`

**Full output?**
```bash
$ stratus detonate aws.exfiltration.ec2-share-ami
2022/03/29 20:14:20 Checking your authentication against the AWS API
2022/03/29 20:14:21 Not warming up - aws.exfiltration.ec2-share-ami is already warm. Use --force to force
2022/03/29 20:14:21 Exfiltrating AMI ami-083ab591a70549402 by sharing it with an external AWS account
2022/03/29 20:14:21 Error while detonating attack technique aws.exfiltration.ec2-share-ami: Unable to share AMI with external AWS account: operation error EC2: ModifyImageAttribute, https response error StatusCode: 400, RequestID: 97354885-357c-497f-b784-3365a5294736, api error InvalidParameter: Snapshots encrypted with the AWS Managed CMK can't be shared. Specify another snapshot.
```

```bash
$ stratus detonate aws.exfiltration.ec2-share-ebs-snapshot
2022/03/29 20:20:23 Checking your authentication against the AWS API
2022/03/29 20:20:23 Not warming up - aws.exfiltration.ec2-share-ebs-snapshot is already warm. Use --force to force
2022/03/29 20:20:23 Sharing the volume snapshot snap-00d459dab53c44042 with an external AWS account...
2022/03/29 20:20:23 Error while detonating attack technique aws.exfiltration.ec2-share-ebs-snapshot: operation error EC2: ModifySnapshotAttribute, https response error StatusCode: 400, RequestID: 12f44aeb-7b3b-4488-ac46-a432d20cc7a9, api error OperationNotPermitted: Encrypted snapshots with EBS default key cannot be shared
```

**Files in $HOME/.stratus-red-team?**
N/A

**Comment by christophetd** [comment]
Thanks for reporting! Does that happen only if the EBS default encryption uses a customer-managed CMK, or also if it uses the AWS default KMS key?

**Comment by christophetd** [comment]
So if we share the EBS snapshot/AMI publicly but it's encrypted with a CMK, it's "public" but an attacker can't access it unless he has explicit access to the CMK for decryption. Correct?

If that's the case, is it really something an attacker would do? (Create a KMS key, grant themselves access to the KMS key, encrypt the snapshot with it, then share it)

I would argue an attacker is likely to _first_ try to share the EBS/AMI publicly. In which case, you could argue that the current implementation makes sense, even if it will fail. Thoughts?

**Comment by mchaffe** [comment]
You made a good point, the `ModifySnapshotAttribute` and `ModifyImageAttribute` events will still be logged even if default encryption prevents them from being successful (just confirmed it). Given there are no further API calls which rely on a successful call, it doesn't matter if they fail. 

So a cleaner approach than modifying terraform would just be to catch the error during detonation. Then include a message that API call failed but detonation was successful.

**Comment by christophetd** [comment]
Sounds great! Thanks for the suggestion.

**Comment by christophetd** [comment]
Looking better in #113 @mchaffe ?

**Comment by christophetd** [comment]
Thanks for the feedback! Will be released as part of v1.6.2, out shortly.
